app/src/main/java/net/gsantner/markor/activity/DocumentShareIntoFragment.java

```diff
@@ -213,6 +216,23 @@ private void appendToExistingDocument(final File file, final String separator, f
             _appSettings.addRecentDocument(file);
         }
 
+        private String formatShare(final String shared) {
+            final Context context = getContext();
+            final String prefix = _appSettings.getShareIntoPrefix();
+            final List<String> parts = new ArrayList<>(Arrays.asList(prefix.split(Pattern.quote(TEXT_TOKEN))));
+
+            // Interpolate parts
+            final long time = System.currentTimeMillis();
+            for (int i = 0 ; i < parts.size(); i++) {
+                parts.set(i, _cu.formatDateTime(context, parts.get(i), time));
+            }
+
+            // Put the shared text in the right place
+            parts.add(1, shared);
+
+            return String.join("", parts);
```

Comment:
Fix share into interpolation / formatting

Reply:
I recall String.join throwing exception on older android devices. There should be some TextUtils methods as alternative around